File: APS/Lincoln_MatheusAPS/Models/Produto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Lincoln_MatheusAPS.Models
{
    public class Produto
    {
        [Key]
        [Display(Name = "Código do Produto")]
        public int IdPd { get; set;}
        [Display(Name = "Nome do Produto")]
        public string? NomeId { get; set;}
        [Display(Name = "Descrição do Produto")]
        public string? Descricao { get; set;}
        [Display(Name = "Quantidade de Produto")]
        public int Quantidade { get; set;}
        [Display(Name = "Preço do Produto")]
        public double Preco { get; set;}

        public List<Item>? Items { get; set;}
        [Display(Name = "Código da Marca")]
        public int IdM { get; set;}
        public Marca? Marca { get; set;}
    }
}

File: Exercicios_Poo/Atividade_07.b/Questao05/Produto.ts
class Produto {
    private _identificador: number;
    private _descricao: string;
    private _quantidadeEstoque: number;
    private _preco: number;

    constructor(identificador: number, descricao: string, quantidadeEstoque: number, preco: number) {
        this._identificador = identificador;
        this._descricao = descricao;
        this._quantidadeEstoque = quantidadeEstoque;
        this._preco = preco;
    }

    get identificador(): number {
        return this._identificador;
    }

    get descricao(): string {
        return this._descricao;
    }

    get quantidadeEstoque(): number {
        return this._quantidadeEstoque;
    }

    get preco(): number {
        return this._preco;
    }

    set identificador(identificador: number) {
        this._identificador = identificador;
    }

    set descricao(descricao: string) {
        this._descricao = descricao;
    }

    set quantidadeEstoque(quantidadeEstoque: number) {
        this._quantidadeEstoque = quantidadeEstoque;
    }

    set preco(preco: number) {
        this._preco = preco;
    }

    public baixarEstoque(quantidade: number): void {
        this._quantidadeEstoque -= quantidade;
    }

    public reporEstoque(quantidade: number): void {
        this._quantidadeEstoque += quantidade;
    }

}

export { Produto };